### PR TITLE
Split resources to separate files

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9174341

### What this PR does / why we need it:

This is no-op preparation for credential rotation so we could re-use these functions when required.

### Test plan for issue:

no-op

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

File: pkg/cluster/deploystorage.go [hunk 1]
```diff
@@ -152,43 +147,11 @@ func (m *manager) deployStorageTemplate(ctx context.Context, installConfig *inst
 		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
 		ContentVersion: "1.0.0.0",
 		Resources: []*arm.Resource{
-			rbac.ResourceGroupRoleAssignmentWithName(
-				rbac.RoleContributor,
-				"'"+clusterSPObjectID+"'",
-				"guid(resourceGroup().id, 'SP / Contributor')",
-			),
```
Comment (troy0820) [inline pkg/cluster/deploystorage.go:159]: nit: function replacing this resource `clusterServicePrincipalRBAC(clusterSPObjectID)` is at the bottom of the slice but does not seem to be a big issue.

File: pkg/cluster/resources.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			},
+			Name:     to.StringPtr("cluster" + m.doc.OpenShiftCluster.Properties.StorageSuffix),
+			Location: &region,
+			Type:     to.StringPtr("Microsoft.Storage/storageAccounts"),
+		},
+		APIVersion: azureclient.APIVersion("Microsoft.Storage"),
+	}
+}
+
+func (m *manager) clusterStorageAccountBlobIgition() *arm.Resource {
```
Comment (jim-minter) [inline pkg/cluster/resources.go:101]:
```suggestion
func (m *manager) clusterStorageAccountBlobIgnition() *arm.Resource {
```

Comment (jim-minter) [inline pkg/cluster/resources.go:101]: could just make "ignition" / "aro" a parameter FWIW - saves function duplication

File: pkg/cluster/resources.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
+package cluster
```
Comment (jim-minter) [inline pkg/cluster/resources.go:1]: any value in calling this file deploystorage_resources.go for symmetry?

**Review by jim-minter — COMMENTED** [review]
couple of nits

**Review by jim-minter — APPROVED** [review]
lgtm, will merge on green

**Pull request merged**